automation/workflow/store.go

```diff
@@ -0,0 +1,307 @@
+package workflow
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/qri-io/qri/profile"
+)
+
+// Store manages & stores workflows, allowing listing and updating of workflows
+type Store interface {
+	Lister
+	// Create creates a new Workflow and adds it to the Store. If a workflow
+	// with that dataset id already exists, it emits an
+	// `ErrWorkflowForDatasetExists` error
+	Create(did string, pid profile.ID, triggers []Trigger, hooks []Hook) (*Workflow, error)
+	Get(wid ID) (*Workflow, error)
+	GetDatasetWorkflow(did string) (*Workflow, error)
+	Remove(id ID) error
+	Update(wf *Workflow) error
```

Comment:
If you're ok with the `Put` change, we can merge update into it & branch off weather the provided workflow has an ID or not (workflow with an ID this store recognizes == update)